tests/integration_tests/test_public_api.py

```diff
@@ -0,0 +1,144 @@
+import ipaddress
+from contextlib import contextmanager, suppress
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Generator, Type
+
+import pytest
+
+import pycloudlib
+from pycloudlib.cloud import BaseCloud
+from pycloudlib.instance import BaseInstance
+from pycloudlib.util import LTS_RELEASES
+
+cloud_config = """\
+#cloud-config
+runcmd:
+  - echo 'hello' > /var/tmp/example.txt
```

Comment:
Can we actually make this append instead of write. This way we can assert cloud-init doesn't run multiple times every boot.
`  - echo 'hello' >> /var/tmp/example.txt`